### PR TITLE
[LOBEMO-481] Add tracers transformers into the node.

### PR DESCRIPTION
**Pull request by ksaric**
https://github.com/input-output-hk/iohk-monitoring-framework/issues/481

We are still missing the `mkOutcome` so we can pass in the traced events into outcomes, calculating the performance on the way.

### CODE REVIEW

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 1]
```diff
@@ -369,6 +394,9 @@ mkTracers traceOptions tracer = do
         $ withName "LocalTxSubmissionProtocol" tracer
       }
 
+--instance Transformable Transformable Text IO (MeasureTxs blk) where
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Tracing/Tracers.hs]: Is this left over?

File: cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs [hunk 2]
```diff
@@ -0,0 +1,122 @@
+{-# LANGUAGE LambdaCase             #-}
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: Please don't align `#-}`

File: cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+deriving instance (ProtocolLedgerView blk, Show blk, Show (GenTx blk)) => Show (MeasureTxs blk)
+
+-- Any Monad m, could be Identity in this case where we have all the data beforehand.
+-- The result of this operation is the list of transactions that _made it in the block_
+-- and the time it took them to get into the block.
+instance (Monad m, Eq (GenTx blk)) => Outcome m (MeasureTxs blk) where
+    type IntermediateValue  (MeasureTxs blk)    = [(GenTx blk, Time)]
+    type OutcomeMetric      (MeasureTxs blk)    = [(GenTx blk, DiffTime)]
+
+    --classifyObservable     :: a -> m OutcomeProgressionStatus
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: Left over comment?

File: cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    classifyObservable = pure . \case
+      MeasureTxsTimeStart {}    -> OutcomeStarts
+      MeasureTxsTimeStop  {}    -> OutcomeEnds
+
+    --captureObservableValue :: a -> m (IntermediateValue a)
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: Left over comment?

File: cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    captureObservableValue (MeasureTxsTimeStart txs _ time) =
+        pure [(tx, time) | tx <- txs]
+
+    captureObservableValue (MeasureTxsTimeStop _sloNo _blk txs time) =
+        pure [(tx, time) | tx <- txs]
+
+    --computeOutcomeMetric   :: a -> IntermediateValue a -> IntermediateValue a -> m (OutcomeMetric a)
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: Left over comment?

File: cardano-node/src/Cardano/Tracing/Tracers.hs (hunk 1 above)
Comment (ksaric) [reply cardano-node/src/Cardano/Tracing/Tracers.hs]: This was removed.

File: cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs (hunk 2 above)
Comment (ksaric) [reply cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: Ah, sorry for aligning things.

File: cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs (hunk 4 above)
Comment (ksaric) [reply cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: No. Intentional.

File: cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs (hunk 5 above)
Comment (ksaric) [reply cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: No. Intentional.

File: cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs (hunk 3 above)
Comment (ksaric) [reply cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: No. Intentional.

File: cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs (hunk 2 above)
Comment (ksaric) [reply cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: Wait. Why do we have it aligned in other modules then?

Comment (Jimbo4350) [reply cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: @ksaric They shouldn't be aligned in other modules. Those files were most likely there before we put in `STYLE.md`

Comment (ksaric) [reply cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: Shouldn't someone fix that then? It's confusing to have a `STYLE.md` which doesn't have styles applied to the code.
The code will be split between two styles.
But fine, I'll remove alignment.

**Comment by ksaric** [comment]
`cardano-cli: ExceptionInLinkedThread ThreadId 10 thread blocked indefinitely in an STM transaction`

**Comment by ksaric** [comment]
@Jimbo4350 Ready for review.

File: cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+-- Any Monad m, could be Identity in this case where we have all the data beforehand.
+-- The result of this operation is the list of transactions that _made it in the block_
+-- and the time it took them to get into the block.
+instance (Monad m, Eq (GenTx blk)) => Outcome m (MeasureTxs blk) where
+    type IntermediateValue  (MeasureTxs blk)    = [(GenTx blk, Time)]
+    type OutcomeMetric      (MeasureTxs blk)    = [(GenTx blk, DiffTime)]
+
+    --classifyObservable     :: a -> m OutcomeProgressionStatus
+    classifyObservable = pure . \case
+      MeasureTxsTimeStart {}    -> OutcomeStarts
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: Alignment here

File: cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+             ]
+
+-- | Transformer for the start of the transaction, when the transaction was added
+-- to the mempool.
+measureTxsStart :: Tracer IO (LogObject Text) -> Tracer IO (TraceEventMempool blk)
+measureTxsStart tracer = measureTxsStartInter $ toLogObject tracer
+  where
+    measureTxsStartInter :: Tracer IO (MeasureTxs blk) -> Tracer IO (TraceEventMempool blk)
+    measureTxsStartInter tracer' = Tracer $ \case
+        TraceMempoolAddTxs txs totalNum time    -> traceWith tracer' (MeasureTxsTimeStart txs totalNum time)
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: Alignment here

File: cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        _                                       -> pure ()
+
+-- | Transformer for the end of the transaction, when the transaction was added to the
+-- block and the block was forged.
+measureTxsEnd :: Tracer IO (LogObject Text) -> Tracer IO (TraceForgeEvent blk (GenTx blk))
+measureTxsEnd tracer = measureTxsEndInter $ toLogObject tracer
+  where
+    measureTxsEndInter :: Tracer IO (MeasureTxs blk) -> Tracer IO (TraceForgeEvent blk (GenTx blk))
+    measureTxsEndInter tracer' = Tracer $ \case
+        TraceAdoptedBlock slotNo blk txs time   -> traceWith tracer' (MeasureTxsTimeStop slotNo blk txs time)
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Tracing/MicroBenchmarking.hs]: Alignment here

**Comment by ksaric** [comment]
Integrated https://github.com/input-output-hk/iohk-monitoring-framework/pull/493 into the PR.

**Pull request merged**